### PR TITLE
Fix unknown speaker errors

### PR DESCRIPTION
**Pull request by Dou1622**
Fix unknown speaker or id errors
See #4494

### CODE REVIEW

**Comment by Pentarctagon** [comment]
As a side question I'm somewhat curious how wmllint determines what a valid speaker is vs isn't, but I'm fine this change as it is regardless of the answer to that.

File: data/campaigns/Winds_of_Fate/scenarios/11_Crosswind.cfg [hunk 1]
```diff
@@ -159,6 +159,7 @@
             [leader_goal]
                 x,y=4,36
                 auto_remove=yes
+                #wmllint: recognize land_on_keep
```
Comment (CelticMinstrel) [inline data/campaigns/Winds_of_Fate/scenarios/11_Crosswind.cfg:162]: This isn't a unit so it's incorrect to use `recognize` on it.

File: data/campaigns/Winds_of_Fate/scenarios/01_The_Hunt.cfg [hunk 2]
```diff
@@ -133,6 +133,7 @@
             [filter]
                 type="Fire Ant"
             [/filter]
+            # wmllint: recognize guarding_nest
```
Comment (CelticMinstrel) [inline data/campaigns/Winds_of_Fate/scenarios/01_The_Hunt.cfg:136]: This also isn't a unit, so `recognize` is incorrect.

**Comment by CelticMinstrel** [comment]
> As a side question I'm somewhat curious how wmllint determines what a valid speaker is vs isn't, but I'm fine this change as it is regardless of the answer to that.

I believe it scans for explicitly-placed units (in `[side]` and `[unit]` tags) and considers those units to be valid. I'm not sure if it also considers `[recall]` tags.

I've noted two IDs that you added `recognize` to that have nothing to do with units. If wmllint is reporting them as invalid speakers, then that is a bug in wmllint and should be fixed in wmllint – it should not analyze IDs nested under AI tags (`[ai]`, `[micro_ai]`, or `[modify_ai]`) unless they are in a tag whose name starts with "filter" (generally `[filter]`, `[filter_second]`, `[filter_own]`, or `[filter_enemy]`).

**Comment by Dou1622** [comment]
> > As a side question I'm somewhat curious how wmllint determines what a valid speaker is vs isn't, but I'm fine this change as it is regardless of the answer to that.
> 
> I believe it scans for explicitly-placed units (in `[side]` and `[unit]` tags) and considers those units to be valid. I'm not sure if it also considers `[recall]` tags.
> 
> I've noted two IDs that you added `recognize` to that have nothing to do with units. If wmllint is reporting them as invalid speakers, then that is a bug in wmllint and should be fixed in wmllint – it should not analyze IDs nested under AI tags (`[ai]`, `[micro_ai]`, or `[modify_ai]`) unless they are in a tag whose name starts with "filter" (generally `[filter]`, `[filter_second]`, `[filter_own]`, or `[filter_enemy]`).

Do IDs should be analyzed in any "filter" tag? Is there a "filter" tag list?

**Comment by Pentarctagon** [comment]
Every filter tag currently available is (I believe) present on the WML tags sidebar on the wiki (ie: on the right side of https://wiki.wesnoth.org/ReferenceWML) under the F section.

**Comment by CelticMinstrel** [comment]
I don't think there's any tag that ends with `_filter`. As far as I know, nearly all filter tags begin with `filter_`. I _can_ think of one exception (other than the obvious `[filter]`): `[goal]` tags (and the other shorthand tags for defining goals) contain a `[criteria]` tag which is a filter (can be either a unit or location filter). Note that only _unit_ filters can contain an ID key, so `[filter_location]` for example isn't a concern.

**Comment by Wedge009** [comment]
I reverted the two instances of non-units, but the rest makes sense to me. It's been long enough, I think this can be merged assuming the checks pass, right?

**Comment by Pentarctagon** [comment]
That sounds fine to me.

**Pull request merged**